src/library_browser.js

```diff
@@ -593,16 +593,12 @@ var LibraryBrowser = {
           Browser.mouseMovementY = Browser.getMovementY(event);
         }
 
-        // check if SDL is available
-        if (typeof SDL != "undefined") {
-          Browser.mouseX = SDL.mouseX + Browser.mouseMovementX;
-          Browser.mouseY = SDL.mouseY + Browser.mouseMovementY;
-        } else {
-          // just add the mouse delta to the current absolut mouse position
-          // FIXME: ideally this should be clamped against the canvas size and zero
-          Browser.mouseX += Browser.mouseMovementX;
-          Browser.mouseY += Browser.mouseMovementY;
-        }
+        // just add the mouse delta to the current absolut mouse position and clamp to surface size
+        Browser.mouseX += Browser.mouseMovementX;
+        Browser.mouseY += Browser.mouseMovementY;
+        var surface = (typeof SDL != 'undefined') ? SDL.surfaces[SDL.screen] : Module['canvas'];
```

Comment:
I had to consider the SDL surface because its size may be smaller than the canvas, and scaled when drawing. In that case, the cursor position would be in the coordinate space of the original SDL surface size rather than the canvas size it's scaled up to.

Reply:
I'm not sure, but that sounds odd to me. If that were the case, we'd need to add special code for SDL2, glfw, etc. as well. But I think it should all be in terms of the absolute actual canvas size, so we don't?